lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -204,7 +204,7 @@ def get_cpu_facts(self, collected_facts=None):
             # model name is for Intel arch, Processor (mind the uppercase P)
             # works for some ARM devices, like the Sheevaplug.
             # 'ncpus active' is SPARC attribute
-            if key in ['model name', 'Processor', 'vendor_id', 'cpu', 'Vendor', 'processor']:
+            if key in ['model name', 'Processor', 'vendor_id', 'Vendor', 'processor']:
```

Comment:
I'm hesitant to remove the `cpu` flag. It does pass our unit tests, but there may be architectures out there relying on `cpu` as a counter that we don't have fixtures for yet.

Reply:
The history in the commit message shows that `cpu` was added explicitly for Power.  If something else is relying thereon (and it's certainly not x86_64, s390x, aarch64, and apparently not sparc64), then it's probably a fluke that it works at all.